From d7cba13e04debd07cc98d361e8bf831917f120c4 Mon Sep 17 00:00:00 2001
From: "Patil, Rachna" <rachna@ti.com>
Date: Tue, 10 Jan 2012 16:38:52 +0530
Subject: [PATCH 0673/1092] input: TSC: Add support for suspend/resume

This patch adds support for suspend and resume
in touchscreen ti_tscadc

Signed-off-by: Patil, Rachna <rachna@ti.com>
---
 drivers/input/touchscreen/ti_tscadc.c | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/drivers/input/touchscreen/ti_tscadc.c b/drivers/input/touchscreen/ti_tscadc.c
index 4a71712..a27fa70 100644
--- a/drivers/input/touchscreen/ti_tscadc.c
+++ b/drivers/input/touchscreen/ti_tscadc.c
@@ -545,6 +545,36 @@ static int __devexit tscadc_remove(struct platform_device *pdev)
 	return 0;
 }
 
+static int tscadc_suspend(struct platform_device *pdev)
+{
+	struct tscadc *ts_dev = platform_get_drvdata(pdev);
+	unsigned int  status;
+
+	status = tscadc_readl(ts_dev, TSCADC_REG_CTRL);
+	status &= ~(TSCADC_CNTRLREG_TSCSSENB);
+	tscadc_writel(ts_dev, TSCADC_REG_CTRL, status);
+
+	clk_disable(ts_dev->tsc_ick);
+	clk_put(ts_dev->tsc_ick);
+
+	return 0;
+
+}
+
+static int tscadc_resume(struct platform_device *pdev)
+{
+	struct tscadc *ts_dev = platform_get_drvdata(pdev);
+	unsigned int  status;
+
+	clk_enable(ts_dev->tsc_ick);
+
+	status = tscadc_readl(ts_dev, TSCADC_REG_CTRL);
+	tscadc_writel(ts_dev, TSCADC_REG_CTRL,
+			(status | TSCADC_CNTRLREG_TSCSSENB));
+
+	return 0;
+}
+
 static struct platform_driver ti_tsc_driver = {
 	.probe	  = tscadc_probe,
 	.remove	 = __devexit_p(tscadc_remove),
@@ -552,6 +582,8 @@ static struct platform_driver ti_tsc_driver = {
 		.name   = "tsc",
 		.owner  = THIS_MODULE,
 	},
+	.suspend = tscadc_suspend,
+	.resume  = tscadc_resume,
 };
 
 static int __init ti_tsc_init(void)
-- 
1.7.11.2

